src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATE_COPY = {
+    ...state,
+  };
+
+  const result = [STATE_COPY];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+    const currentStep = {
+      ...result[result.length - 1],
+    };
```

Comment:
try to simplify the solution by not creating this variable :)

Reply:
done